File: rolemadelen/programmers/120842_2차원으로_만들기.cpp
#include <vector>
using namespace std;

vector<vector<int>> solution(vector<int> num_list, int n) {
    vector<vector<int>> ans;
    
    for(int i=0; i<num_list.size(); i+=n) {
        vector<int> temp;
        for(int j=0; j<n; ++j) {
            temp.push_back(num_list[i+j]);
        }
        ans.push_back(temp);
    }
    return ans;
}

File: simsu/2주차/120842_2차원으로_만들기.js
function solution(num_list, n) {
    let answer = Array.from(Array(Math.floor(num_list.length/n)), () => new Array(n) );
    let y = 0;
    let x = 0;
    num_list.forEach( val => {
        if(y==n) {
            y = 0;
            x++;
        }
        answer[x][y] = val;
        y++;
    })
    return answer;
}